### ISSUE TITLE
Add setting of Enclosure AmpMinimumLevel to the GUI

### ISSUE THREAD

**Issue by jason-barba** [GrandOrgue/grandorgue#782]
The Enclosure minimum amplitude level should be settable from the GUI. This is IMHO a parameter that is more related to the local hardware (like amplifier, speakers etc) and the environment of the VPO than to the sample set.

//Jens

**Comment by jason-barba** [comment]
Reference #769

**Comment by mutti-wilson** [comment]
I agree. This is something I adjust on sample sets to fit the room or just personal taste. Since it is only a volume adjustment and not changing EQ or other parameters, modifying it has no effect on the overall sound quality.

**Comment by mutti-wilson** [comment]
Check out this that could possibly change this all up.

https://github.com/GrandOrgue/grandorgue/issues/717

**Comment by oleg68** [comment]
There are two kinds of organ settings GUI now:

- Organ settings dialog. It mostly designed for changing pipe properties in the hierarchy (pipe - rank - windchest - organ)
- Midi settings. It is opened with a right click and is designed for settings MIDI properties.

Thinking on implementing setting an enclosure minimum level and another properties in the future, we may add it either to the Organ Settings dialog or to the MIDI settings dialog.

I would prefer to add it to the MIDI settings dialog to an additional tab. My reasons:
- MIDI settings dialog already has tabs, so adding a new tab would not change the GUI.
- There is no any enclosure hierarchy to add to the Organ Settings. Now enclosures are not shown in this dialog.
- Right-clicking is more intuitive for the user than opening a separate dialog

But there are also some contras:
- Minimum amplitude is not related to MIDI directly. So we should rename the dialog from "MIDI settings" to "Settings"
- There won't be a visual relationship between the enclosure and the windchest.

@larspalo @rousseldenis what are your opinions on where we should add this setting?

**Comment by oleg68** [comment]
> it's an ODF specific operation that the user shouldn't be allowed to change - it's the responsibility for the sample set designer (organ builder...).

What is the difference between AmpMinLevel and Ampllitude/Gain, PitchTuning of pipes? Why the second should be customisable and the first should not.

**Comment by larspalo** [comment]
> Why the second should be customisable and the first should not?

Because for "mistakes" done on the pipes by the sample set producer there's no real alternative! For enclosures there's already built in versions supplied by GO itself to use instead as alternatives. It's the GO built in versions that should be possible to customize the desired low limit of for the user.

Currently all the built in enclosures will always bottom out at 0, which will kill the sound completely, which some may indeed want especially for the global organ level (master). If the low limit of all the built in enclosures could be customized individually, everything would be possible to do for an user by using them instead of the organ defined ones (which might not even exist at all).

**Comment by Psi58** [comment]
I agree that users should respect the sampleset designer's choices wherever possible, otherwise the unique characteristics of a particular organ would be lost. 

Practically, however, we play organs on imperfect consoles, in spaces that are acoustically totally unlike the original organ's location. So users need to tweak the audio to suit their individual environment.

In my personal experience, using the tone modifications makes organ voicing sound better in my small room. And an external graphic equaliser helps to dial down output at the resonant frequency. I also often reduce the enclosures AmpMinimumLevel as there is insufficient dynamic range from the enclosure output otherwise. In none of those cases am I trying to make the organ sound significantly different from the original - just to make it work in my environment.

So I still support the original request, for these reasons. It doesn't matter to me which settings menu the control appears in - the important thing is that regular users shouldn't need to edit the odf.

**Comment by larspalo** [comment]
Seems like you're all missing the point - there are already built in enclosures that GO provide for _every_ windchest. Currently they are forced to have an AmpMinLevel of 0. This is THE value that should be configurable, NOT the values of any odf defined enclosure.

**Comment by Psi58** [comment]
Highly likely! Happy to acknowledge my very rudimentary knowledge - I rely on help materials for everything I tinker with, and know just enough to be dangerous...
So... I have an ODF with an enclosure AMPMinimumLevel=30. Are there instructions anywhere, either written or video, for overriding this with values from the built-in GO enclosure? Or am I still misunderstanding?

**Comment by larspalo** [comment]
> Are there instructions anywhere, either written or video, for overriding this with values from the built-in GO enclosure? Or am I still misunderstanding?

You just connect your MIDI swell pedal to the appropriate built-in enclosure on the "Coupler manuals and volume" panel instead of connecting to the swell pedal (enclosure) that the ODF provides (even if the ODF doesn't provide any at all, this panel will always provide a master volume and swell pedals for every windchest anyway).

![Image](https://github.com/user-attachments/assets/d50658c3-5481-4abc-94d5-f86889ef3a8e)

Currently the min level of a built-in enclosure is not possible to customize but I think that it should be...

**Comment by Psi58** [comment]
Thanks again - works as you described on one of my organs.
But on another : 

![Image](https://github.com/user-attachments/assets/a596de65-9902-4178-8b9b-5eb0e6f478f2)

No expression pedals! This is a HW sampleset that was ODFedit-ed, so I don't know if the behaviour is caused by the original HW xml code, or the ODFedit process.

**Comment by larspalo** [comment]
> No expression pedals!

Well, let me guess, there are a lot of windchests in that sample set! And another guess is that there's another bug in GO that causes problems with the layout of the swell pedals when there are a lot of them...

**Comment by Psi58** [comment]
Spot on - there are 12.
I see you've already diagnosed the issue in https://github.com/GrandOrgue/grandorgue/issues/2100
